stonesoup/dataassociator/general.py

```diff
@@ -0,0 +1,167 @@
+
+        distance_matrix = np.empty((len(objects_a), len(objects_b)))
+
+        list_of_as = list(objects_a)
+        list_of_bs = list(objects_b)
+
+        for i, a in enumerate(list_of_as):
+            for j, b in enumerate(list_of_bs):
+                distance_matrix[i, j] = self.individual_weighting(a, b)
+
+        # Use "shortest path" assignment algorithm on distance matrix
+        # to assign tracks to nearest detection
+        # Maximise flag = true for probability instance
+        # (converts minimisation problem to maximisation problem)
+        row_ind, col_ind = linear_sum_assignment(
+            distance_matrix, self.maximise_measure)
+
+        # Create dictionary for associations
+        associations = AssociationSet()
+
+        # Generate dict of key/value pairs
+        for i, j in zip(row_ind, col_ind):
+            object_a = list_of_as[i]
+            object_b = list_of_bs[j]
+
+            value = distance_matrix[i, j]
+
+            # Check association meets threshold
+            if self.maximise_measure:
+                if value > self.association_threshold:
+                    # Meets threshold
+                    associations.associations.add(Association({object_a, object_b}))
+            else:  # Minimise measure
+                if value < self.association_threshold:
+                    # Meets threshold
+                    associations.associations.add(Association({object_a, object_b}))
+
+        associated_objects = {obj
+                              for assoc in associations.associations
+                              for obj in assoc.objects}
+
+        unassociated_a = set(objects_a) - associated_objects
+        unassociated_b = set(objects_b) - associated_objects
+
+        return associations, unassociated_a, unassociated_b
+
+    @property
+    def fail_value(self):
+        """
+        For an association to be valid is must be over (or under if maximise_measure is True)
+        (non-inclusive). Therefore setting the value to the association threshold will result in
+        the association not taking place.
+        """
+        return self.association_threshold
+
+    def individual_weighting(self, a, b):
+        """ This wrapper around the measure function allows for filtering/error checking of the
+        measure function. It can give an easy access point for subclasses that want to apply
+        additional filtering or gating."""
+        measure_output = self.measure(a, b)
+        if measure_output is None:
+            return self.fail_value
+        else:
+            if self.maximise_measure:
+                return max(measure_output, self.fail_value)
+            else:
+                return min(measure_output, self.fail_value)
+
+    def association_dict(self, objects_a: Collection, objects_b: Collection) -> dict:
+        """
+        This is a wrapper function around the `associate` function. The two collections of objects
```

Comment:
```suggestion
        This is a wrapper function around :func:`associate`. The two collections of objects
```
Or something similar if that doesn't work.